### PR TITLE
Publish rules_haskell@0.18

### PR DESCRIPTION
**Pull request by avdv**

### CODE REVIEW

**Comment by avdv** [comment]
Is there a way to have the CI pipeline use Bazel 6.x instead of Bazel 7?

**Comment by fmeum** [comment]
Yes, see https://github.com/bazelbuild/continuous-integration/blob/master/buildkite/README.md#using-a-specific-version-of-bazel

File: modules/rules_haskell/0.18/MODULE.bazel [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module(
     name = "rules_haskell",
     version = "0.18",
+    bazel_compatibility = [">=6.0.0", "<=6.99.0"],
```
Comment (avdv) [inline modules/rules_haskell/0.18/MODULE.bazel:4]: The design doc just mentioned three possible comparison operators, but I see that you also can use `<`.

Comment (avdv) [reply modules/rules_haskell/0.18/MODULE.bazel]: I did remove the bazel_compatibility setting again, it should be applied to rules_sh@0.3.0 IMO

**Review by fmeum — APPROVED** [review]

**Pull request merged**